src/main/drivers/max7456.c

```diff
@@ -571,14 +571,18 @@ uint8_t* max7456GetScreenBuffer(void)
 
 void max7456WriteChar(uint8_t x, uint8_t y, uint8_t c)
 {
-    screenBuffer[y*CHARS_PER_LINE+x] = c;
+    if (y < VIDEO_LINES_PAL) {
+        screenBuffer[y*CHARS_PER_LINE+x] = c;
+    }
 }
 
 void max7456Write(uint8_t x, uint8_t y, const char *buff)
 {
-    for (int i = 0; *(buff+i); i++) {
-        if (x+i < CHARS_PER_LINE) {// Do not write over screen
-            screenBuffer[y*CHARS_PER_LINE+x+i] = *(buff+i);
+    if (y < VIDEO_LINES_PAL) {
+        for (int i = 0; *(buff+i); i++) {
+            if (x+i < CHARS_PER_LINE) {// Do not write over screen
+                screenBuffer[y*CHARS_PER_LINE+x+i] = *(buff+i);
+            }
```

Comment:
You could flip this around and do an early return if the position exceeds the column width - no point continuing to loop to the end of the string. For example:
```
    for (int i = 0; *(buff + i); i++) {
        if (x + i >= CHARS_PER_LINE) {
            return;
        }
        screenBuffer[y * CHARS_PER_LINE + x + i] = *(buff + i);
    }
```
Also the formatting could be cleaned up (spaces around operators added) - same with the `max7456WriteChar()` function.